### PR TITLE
Np 8654 search project status

### PR DESCRIPTION
**Pull request by svenngjeto**
Search for Projects with additional query parameter 'status'. Allowed input values are ACTIVE, CONCLUDED and  NOTSTARTED. Query parameter 'query' is still required.

### CODE REVIEW

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/FetchCristinProjects.java [hunk 1]
```diff
@@ -62,7 +67,13 @@ protected FetchCristinProjects(CristinApiClient cristinApiClient, Environment en
         if (requestInfo.getQueryParameters().containsKey(ORGANIZATION)) {
             requestQueryParameters.put(ORGANIZATION, getValidOrganizationUri(requestInfo));
         }
+        Optional<ProjectStatus> projectStatus = getValidProjectStatus(requestInfo);
```
Comment (brinxmat) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/FetchCristinProjects.java:70]: Can you use a declarative ifPresent?

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/FetchCristinProjects.java [hunk 2]
```diff
@@ -92,4 +103,15 @@ protected Integer getSuccessStatusCode(Void input, SearchResponse output) {
         return requestQueryParameters;
     }
 
+    private Optional<ProjectStatus> getValidProjectStatus(RequestInfo requestInfo) throws BadRequestException {
+        if (requestInfo.getQueryParameters().containsKey(STATUS)) {
+            return Optional.ofNullable(attempt(() ->
+                    ProjectStatus.getNvaStatus(requestInfo.getQueryParameters().get(STATUS)))
+                    .orElseThrow(fail -> new BadRequestException(
+                            invalidQueryParametersMessageWithRange(STATUS, Arrays.toString(ProjectStatus.values())))));
```
Comment (brinxmat) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/FetchCristinProjects.java:111]: Extract so this does one thing: finds valid statuses

File: cristin-project/src/main/java/no/unit/nva/cristin/projects/ProjectStatus.java [hunk 3]
```diff
@@ -7,28 +7,52 @@
...
+     * Maps to a ProjectStatus from given string.
+     *
+     * @param name string to map
+     * @return corresponding ProjectStatus in NVA
+     */
+    public static ProjectStatus getNvaStatus(String name) {
+        return stream(values())
+                .filter(nameType -> nameType.name().equalsIgnoreCase(name))
+                .findFirst()
+                .get();
```
Comment (brinxmat) [inline cristin-project/src/main/java/no/unit/nva/cristin/projects/ProjectStatus.java:50]: I'd be tempted by a Commons::SingletonCollector that throws if the input is not a member

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/projectQuery.feature [hunk 4]
```diff
@@ -160,16 +162,34 @@ Feature: API tests for Cristin projects query
...
     Given path '/project/'
-    * def organizationId = 'htttps:/api.dev.nva.aws.unit.no/cristin/organization/20202.0.0.0'
+    * def illegalOrganizationId = 'htttps:/api.dev.nva.aws.unit.no/cristin/organization/20202.0.0.0'
     And param query = queryString
-    And param organization = organizationId
+    And param organization = illegalOrganizationId
     When method GET
     Then status 400
+
+  Scenario Outline: Query accepts status parameter value case independent
```
Comment (brinxmat) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/projectQuery.feature:178]: What happens if the query contains > 1 instances of the parameter?

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by torarnet — DISMISSED** [review]

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/projectQuery.feature (hunk 4 above)
Comment (svenngjeto) [reply cristin-project/src/test/java/no/unit/nva/cristin/projects/projectQuery.feature:178]: Last parameter is used..  RequestInfo uses HashMap to keep parameters

**Review by brinxmat — APPROVED** [review]

**Pull request merged**